DFC.ServiceTaxonomy.GraphSync/GraphSyncers/GraphSyncHelper.cs

```diff
@@ -61,38 +61,50 @@ public async Task<IEnumerable<string>> NodeLabels()
         {
             CheckPreconditions();
 
-            string nodeLabel = await TransformOrDefault(GraphSyncPartSettings!.NodeNameTransform, _contentType!);
-
-            return new[] {nodeLabel, CommonNodeLabel};
+            return await NodeLabels(GraphSyncPartSettings!, _contentType!);
         }
 
         public async Task<IEnumerable<string>> NodeLabels(string contentType)
         {
             var graphSyncPartSettings = GetGraphSyncPartSettings(contentType);
 
-            string nodeLabel = await TransformOrDefault(graphSyncPartSettings.NodeNameTransform, contentType);
+            return await NodeLabels(graphSyncPartSettings, contentType);
+        }
+
+        private async Task<IEnumerable<string>> NodeLabels(GraphSyncPartSettings graphSyncPartSettings, string contentType)
+        {
+            string nodeLabel = graphSyncPartSettings.NodeNameTransform switch
+            {
+                "$\"ncs__{ContentType}\"" => $"ncs__{contentType}",
+                "$\"esco__{ContentType}\"" => $"esco__{contentType!}",
+                _ => await TransformOrDefault(graphSyncPartSettings.NodeNameTransform, contentType, contentType)
```

Comment:
Out of interest more than anything else, what does this do?

Reply:
The switch expression checks for the actual `NodeNameTransform` config values we use for NCS and ESCO nodes, and if it finds them, it creates the node label without invoking Roslyn. This is to speed up the import process.
If the transform is not one of the current values we use in the startup recipe, it executes the transform code snippet to generate the node label.